### PR TITLE
feat: Add `Expr.round` to Dask

### PR DESCRIPTION
**Pull request by mistShard**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [x] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #637 
- Closes #

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [ ] Documented the changes

### CODE REVIEW

**Comment by lucianosrp** [comment]
I changed the title to avoid ambiguity with #696

**Review by MarcoGorelli — APPROVED** [review]
thanks @mistShard !

File: narwhals/_dask/expr.py [hunk 1]
```diff
@@ -244,6 +244,12 @@ def sum(self) -> Self:
             "sum",
         )
 
+    def round(self) -> Self:
+        return self._from_call(
+            lambda _input: _input.round(),
+            "round",
+        )
```
Comment (MarcoGorelli) [inline narwhals/_dask/expr.py:251]: ah you're missing a `decimals` argument

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: narwhals/_dask/expr.py (hunk 1 above)
Comment (mistShard) [reply narwhals/_dask/expr.py:251]: @MarcoGorelli  I see that now. I would have submitted a resolution sooner but my government basically shutdown the internet for a view days.

Comment (mistShard) [reply narwhals/_dask/expr.py]: > ah you're missing a `decimals` argument

I think  i have resolved it in my new push

File: narwhals/_dask/expr.py [hunk 2]
```diff
@@ -244,6 +244,12 @@ def sum(self) -> Self:
             "sum",
         )
 
+    def round(self, decimal: int = 0) -> Self:
+        return self._from_call(
+            lambda _input: _input.round(decimal),
+            "round",
+        )
```
Comment (FBruzzesi) [inline narwhals/_dask/expr.py:251]: As we need to pass args and kwargs to `_from_call`, there are a couple of nitty gretty details to take care of.
You can check how this is accomplished in `is_between` a few lines above

**Review by FBruzzesi — COMMENTED** [review]
Hey thanks for addressing the issue. 

I left a comment as some more changes might be needed.
Also we merged a lot of changes in the tests folder, which leads to merge conflicts sadly. Try to resolve them and incorporate the test in the general round test

File: narwhals/_dask/expr.py (hunk 2 above)
Comment (mistShard) [reply narwhals/_dask/expr.py:251]: Looking at `is_between` I altered the code to:
```
def round(self, decimals: int = 0) -> Self:
    return self._from_call(
        lambda _input, decimals: _input.round(decimals),
        "round",
        decimals
    )
```
is this more accurate?

Comment (FBruzzesi) [reply narwhals/_dask/expr.py:251]: Yep that looks on point! 🚀👌

Comment (mistShard) [reply narwhals/_dask/expr.py]: @FBruzzesi I've looked into it, but I am not clear on what exactly the additional parameter is supposed to do. Could you please clarify it for me if you can?

**Pull request closed without merging**